dll/win32/browseui/CAutoComplete.cpp

```diff
@@ -25,664 +25,2124 @@
+{
+    TRACE("CACEditCtrl::OnKillFocus(%p)\n", this);
+    ATLASSERT(m_pDropDown);
 
-    *pdwFlag = m_options;
+    // hide the list if lost focus
+    HWND hwndGotFocus = (HWND)wParam;
+    if (hwndGotFocus != m_hWnd && hwndGotFocus != m_pDropDown->m_hWnd)
+    {
+        m_pDropDown->HideDropDown();
+    }
 
-    return hr;
+    bHandled = FALSE; // do default
+    return 0;
 }
 
-/**************************************************************************
- *  IAutoComplete_fnSetOptions
- */
-HRESULT WINAPI CAutoComplete::SetOptions(DWORD dwFlag)
+// WM_PASTE @implemented
+// An application sends this message to an edit control to paste at the current caret position.
+LRESULT CACEditCtrl::OnPaste(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
 {
-    HRESULT hr = S_OK;
-
-    TRACE("(%p) -> (0x%x)\n", this, dwFlag);
-
-    m_options = (AUTOCOMPLETEOPTIONS)dwFlag;
+    TRACE("CACEditCtrl::OnPaste(%p)\n", this);
+    ATLASSERT(m_pDropDown);
 
-    if ((m_options & ACO_AUTOSUGGEST) && m_hwndEdit && !m_hwndListBox)
-        CreateListbox();
+    LRESULT ret = DefWindowProcW(uMsg, wParam, lParam); // do default
+    m_pDropDown->OnEditUpdate(TRUE);
+    return ret;
+}
 
-    return hr;
+// WM_SETFOCUS
+// This message is sent to a window after it has gained the keyboard focus.
+LRESULT CACEditCtrl::OnSetFocus(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
+{
+    TRACE("CACEditCtrl::OnSetFocus(%p)\n", this);
+    ATLASSERT(m_pDropDown);
+    bHandled = FALSE; // do default
+    return 0;
 }
 
-/* Edit_BackWord --- Delete previous word in text box */
-static void Edit_BackWord(HWND hwndEdit)
+// WM_SETTEXT
+// An application sends this message to set the text of a window.
+LRESULT CACEditCtrl::OnSetText(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
 {
-    INT iStart, iEnd;
-    iStart = iEnd = 0;
-    SendMessageW(hwndEdit, EM_GETSEL, (WPARAM)&iStart, (LPARAM)&iEnd);
+    TRACE("CACEditCtrl::OnSetText(%p)\n", this);
+    ATLASSERT(m_pDropDown);
+    if (!m_pDropDown->m_bInSetText)
+        m_pDropDown->HideDropDown(); // it's mechanical WM_SETTEXT
+    bHandled = FALSE; // do default
+    return 0;
+}
 
-    if (iStart != iEnd || iStart < 0)
-        return;
+//////////////////////////////////////////////////////////////////////////////
+// CACListView
 
-    size_t cchText = GetWindowTextLengthW(hwndEdit);
-    if (cchText < (size_t)iStart || (INT)cchText <= 0)
-        return;
+CACListView::CACListView() : m_pDropDown(NULL), m_cyItem(CY_ITEM)
+{
+}
 
-    CComHeapPtr<WCHAR> pszText;
-    if (!pszText.Allocate(cchText + 1))
-        return;
+HWND CACListView::Create(HWND hwndParent)
+{
+    ATLASSERT(m_hWnd == NULL);
+    DWORD dwStyle = WS_CHILD | /*WS_VISIBLE |*/ WS_CLIPSIBLINGS | LVS_NOCOLUMNHEADER |
+                    LVS_OWNERDATA | LVS_OWNERDRAWFIXED | LVS_SINGLESEL | LVS_REPORT;
+    HWND hWnd = ::CreateWindowExW(0, GetWndClassName(), L"Internet Explorer", dwStyle,
```

Comment:
Internet Explorer?! Windows really does that?

Reply:
I can confirm it by WinHier. This is the evidence:
![evidence](https://user-images.githubusercontent.com/2107452/110414940-44c33d00-80d4-11eb-83e6-495e64c9b01e.png)